crates/meilisearch/tests/batches/mod.rs

```diff
@@ -1021,11 +1023,12 @@ async fn test_summarized_batch_cancelation() {
     let server = Server::new().await;
     let index = server.index("doggos");
     // to avoid being flaky we're only going to cancel an already finished batch :(
-    index.create(None).await;
-    index.wait_task(0).await;
-    server.cancel_tasks("uids=0").await;
-    index.wait_task(1).await;
-    let (batch, _) = index.get_batch(1).await;
+    let (task,_status_code) = index.create(None).await;
+    index.wait_task(task.uid()).await.succeeded();
+    let (task,_status_code) = server.cancel_tasks("uids=0").await;
+    index.wait_task(task.uid()).await.succeeded();
+    //TODO: create a get_batch function interface that accepts u64, and remove the following cast.
+    let (batch, _) = index.get_batch(task.uid().to_u32().unwrap()).await;
```

Comment:
Hey, you're right that it would be better to retrieve the actual batch used in the task.
But we should definitely not use the task uid as a batch uid here.
Either we keep the previous version which was using a `1` or if you have the time you could add a method very similar to the `.uid()` but named `.batch_uid()` that returns the batch uid.
Feel free to also change the signature of the `get_batch` function so it can accept whatever you want, it's not really important. 
In the Meilisearch code it cannot accept anything else than a `u32` though.